app/src/main/java/com/android/shelfLife/ui/recipes/execution/FoodItemSelector.kt

```diff
@@ -0,0 +1,222 @@
+
+import android.util.Log
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.ArrowBack
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.FloatingActionButton
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
+import androidx.compose.material3.TopAppBar
+import androidx.compose.material3.TopAppBarDefaults
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.unit.dp
+import androidx.hilt.navigation.compose.hiltViewModel
+import com.android.shelfLife.model.newFoodItem.FoodItem
+import com.android.shelfLife.ui.navigation.LIST_TOP_LEVEL_DESTINATION
+import com.android.shelfLife.ui.navigation.NavigationActions
+import com.android.shelfLife.ui.navigation.Route
+import com.android.shelfLife.ui.newnavigation.BottomNavigationMenu
+import com.android.shelfLife.viewmodel.recipes.ExecuteRecipeViewModel
+import androidx.compose.material.icons.filled.Close
+
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun SelectFoodItemsForIngredientScreen(
+    navigationActions: NavigationActions,
+    viewModel: ExecuteRecipeViewModel = hiltViewModel(),
+    onNext: () -> Unit,
+    onPrevious: () -> Unit
+) {
+    val ingredientName by viewModel.currentIngredientName.collectAsState()
+    val availableFoodItems by viewModel.foodItems.collectAsState()
+    val selectedMap by viewModel.selectedFoodItemsForIngredients.collectAsState()
+    val currentlySelectedItems = selectedMap[ingredientName] ?: emptyList()
+
+    Log.d("SelectFoodItemsScreen", "Current ingredient: $ingredientName")
+    Log.d(
+        "SelectFoodItemsScreen",
+        "Available food items: ${availableFoodItems.joinToString { it.foodFacts.name }}"
+    )
+    Log.d(
+        "SelectFoodItemsScreen",
+        "Currently selected items: $currentlySelectedItems"
+    )
+
+    if (ingredientName == null) {
+        Log.e("SelectFoodItemsScreen", "Ingredient name is null. Navigating back.")
+        navigationActions.goBack()
+        return
+    }
+
+    Scaffold(
+        topBar = {
+            TopAppBar(
```

Comment:
Use Custom top app bar

Reply:
the custom bar doesnt allow me to use other icons such as the Icons.Default.Close, which i need.